lib/modules/datasource/azure-pipelines-tasks/index.ts

```diff
@@ -39,8 +85,49 @@ export class AzurePipelinesTasksDatasource extends Datasource {
     key: (url: string) => url,
     ttlMinutes: 24 * 60,
   })
-  async getTasks(url: string): Promise<Record<string, string[]>> {
-    const { body } = await this.http.getJson<Record<string, string[]>>(url);
+  async getTasks<ResT, Schema extends ZodType<ResT> = ZodType<ResT>>(
+    url: string,
+    opts: HttpOptions,
+    schema: Schema,
+  ): Promise<TypeOf<Schema>> {
+    const { body } = await this.http.getJson(url, opts, schema);
     return body;
   }
+
+  @cache({
+    namespace: `datasource-${AzurePipelinesTasksDatasource.id}`,
+    key: (url: string) => url,
+    ttlMinutes: 24 * 60,
+  })
+  async getFallbackTasks<ResT>(url: string): Promise<ResT> {
+    const { body } = await this.http.getJson<ResT>(url);
```

Comment:
Our http functions accept schema object, so internally it will check against it and return the proper type, so you don't have to specify generic

Reply:
I was going to do it but I don't know how to convert Record<string, string[]> to a schema.

That code you mention is not from this PR, we are talking about the fallback that already existed: https://github.com/renovatebot/azure-devops-marketplace/blob/main/azure-pipelines-builtin-tasks.json

I have simply split the methods in 2 so that “the new” uses schemas.

Reply:
Use `z.record(z.string().array())`

Reply:
Done, thanks!